linkcheck/htmlutil/loginformsearch.py

```diff
@@ -41,7 +41,7 @@ def search_form(content, cgiuser, cgipassword):
     CGI fields. If no form is found return None.
     """
     soup = htmlsoup.make_soup(content)
-    cginames = {cgiuser, cgipassword}
+    cginames = {x for x in (cgiuser, cgipassword) if x is not None}
```

Comment:
It's a set, so we can have fun with set operations!
```suggestion
    cginames = {cgiuser, cgipassword} - {None}
```

I'm not sure that reads better, but it's fun!

Reply:
When `cgiuser is cgipassword is None`, this function will return the first found form (because the empty set is a subset of any other set).  That is not _wrong_ per se, but is it useful?

Reply:
Ah, the caller currently ensures either `cgiuser` or `cgipassword` will not be None.

Reply:
> I'm not sure that reads better, but it's fun!
It is a bit of a mind-bender but I do like it, done.

Reply:
> Ah, the caller currently ensures either `cgiuser` or `cgipassword` will not be None.

I did think about putting more checks in here, but as search_form() was only used in one place and there are those checks in visit_loginurl() and more in configuration/__init__.py I thought it was just adding adding overhead that would never be useful.